tests/resources/test_design_space.py

```diff
@@ -39,3 +45,56 @@ def test_design_space_build():
     assert design_space.uid == design_space_id
     assert design_space.name == 'My Design Space'
     assert design_space.dimensions[0].descriptor.key == 'foo'
+
+
+def test_design_space_limits():
+    """Test that the validation logic is triggered before post/put-ing enumerated design spaces."""
+    # Given
+    session = FakeSession()
+    collection = DesignSpaceCollection(uuid.uuid4(), session)
+
+    too_long = EnumeratedDesignSpace(
+        "foo",
+        "bar",
+        descriptors=[RealDescriptor("x", 0, 1)],
+        data=[{"x": random()} for _ in range(2001)]
+    )
+
+    too_wide = EnumeratedDesignSpace(
+        "foo",
+        "bar",
+        descriptors=[RealDescriptor("R-{}".format(i), 0, 1) for i in range(129)],
+        data=[]
+    )
+
+    just_right = EnumeratedDesignSpace(
+        "foo",
+        "bar",
+        descriptors=[RealDescriptor("R-{}".format(i), 0, 1) for i in range(10)],
+        data=[{"R-{}".format(i): random() for i in range(128)} for _ in range(1000)]
+    )
+
+    # create mock post response by setting the status
+    mock_response = just_right.dump()
+    mock_response["status"] = "READY"
+    mock_response["experimental"] = True
+    mock_response["experimental_reasons"] = ["This is a test", "That experimental reasons work"]
```

Comment:
Is there a more natural place to test these? Something that's testing module building from expected json?

Reply:
I think these just bled through (and I moved them to a difference place in a different PR).  I'll remove them.